### ISSUE TITLE
[BUG]: Crash on Linux when closing the emulator

### ISSUE THREAD

**Issue by NetSysFire** [PCSX2/pcsx2#5503]
### Describe the Bug

When you Alt+F4 the window, the emulator crashes.

### Reproduction Steps

1. Run any game.
2. Alt+F4 the window.
3. The game will freeze briefly.
4. Crash.

```
Interface is initializing.  Entering Pcsx2App::OnInit!
Applying operating system default language...
Command line parsing...
Command line parsed!
zsh: trace trap (core dumped)  pcsx2
```

This does not happen if I run the emulator in the terminal and `^C` it or send it a SIGTERM via `killall`, this specifically has to be done via my window manager (which is awesomewm on Xorg, did not test any other WM oder DE since I have none installed). Just make the window close graphically (either alt+f4 or click on the X on top of the window), which is always a graceful kill.
Just closing the window while no game is running works perfectly fine.

This is a regression, this did not happen in previous versions. In fact, I think I noticed it *around* 1.7.2277. No bisecting done yet.
There is nothing in the logs the emulator writes. See above for the output I got when running it in the terminal. Trace taken from my journal: 
[trace.txt](https://github.com/PCSX2/pcsx2/files/8027394/trace.txt)

### PCSX2 Revision

1.7.2351

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Arch Linux

**Comment by F0bes** [comment]
Can you try `pcsx2-git` on the aur.
The official pcsx2 package has been known to be questionable.

**Comment by F0bes** [comment]
Usually the first step for providing support is advising users to use the aur package. For whatever reason it has solved issues. I personally haven't looked at their packaging process so I don't know what exactly is different.
They also provided a pretty incomplete 64 bit build against upstream's (us) wishes. But that's old news now.

Whenever you have time, just bisect and someone can look into the commit breaking things.

**Comment by NetSysFire** [comment]
Bisecting done. Git says the first bad commit is aaed932a6781264025baa32f4bb56f2cf5146540:

```
commit aaed932a6781264025baa32f4bb56f2cf5146540 (tag: v1.7.2185, refs/bisect/bad)
Author: TellowKrinkle <tellowkrinkle@gmail.com>
Date:   Thu Dec 2 03:03:00 2021 -0600

    PAD:SDL: Remove fallback code for unsupported SDL versions
```

**Comment by tellowkrinkle** [comment]
Okay two things
1. The comment said SDL 2.0.4, but we were checking SDL_MINOR_VERSION, which corresponds to the `0`, not the `4`, so it was always disabling the code
2. Clearly, SDL versions greater than 2.0.4 are affected

...it would be nice to figure out why SDL is crashing
If you could link to a debug build of SDL2 and get a backtrace, that would be nice

**Comment by NetSysFire** [comment]
So, uh, I used gdb and as soon as I started a game while running pcsx2 in gdb, the emulator segfaulted. It looks like this may be a different, unrelated issue but I am not sure how I should proceed when I can't run it in gdb to obtain the trace for the actual freeze I am after.

[trace.log](https://github.com/PCSX2/pcsx2/files/8101162/trace.log)

The curious thing is I can start games and also reproduce the freeze this issue actually is about without gdb.

**Comment by NetSysFire** [comment]
Oooh, you want me to build pcsx2 with debug symbols, too. I only built sdl2-git with that. My bad. Its my first time doing all this in-depth debugging.

**Comment by F0bes** [comment]
> So, uh, I used gdb and as soon as I started a game while running pcsx2 in gdb, the emulator segfaulted. It looks like this may be a different, unrelated issue but I am not sure how I should proceed when I can't run it in gdb to obtain the trace for the actual freeze I am after.
> 
> [trace.log](https://github.com/PCSX2/pcsx2/files/8101162/trace.log)
> 
> The curious thing is I can start games and also reproduce the freeze this issue actually is about without gdb.
> 
> For clarification, this is a freshly built version of pcsx2, so this is the latest master.

In gdb, before you run pcsx2 type
`handle SIGSEGV nonstop noprint`
The recompiler uses segfaults to detect when a block is dirty / needs to be recompiled.

**Comment by NetSysFire** [comment]
Alright. I *think* I did it right this time, the trace looks a lot more verbose.
[trace.log](https://github.com/PCSX2/pcsx2/files/8101285/trace.log)

I sent SIGINT when pcsx2 froze upon me alt+f4'ing the window and then did `thread apply all bt full`.

**Comment by lightningterror** [comment]
Test #5544 and see if the issue is resolved.

**Comment by NetSysFire** [comment]
Unfortunately it seems it does not resolve this. Here is a new trace: 
[trace.log](https://github.com/PCSX2/pcsx2/files/8103382/trace.log)

**Comment by tellowkrinkle** [comment]
So it still crashes *and* freezes or only freezes now?

**Comment by NetSysFire** [comment]
I did not perceive any change. To be clear, the current behavior is that it "just" freezes. It freezed *and* crashed in the past and I also saw a commit while bisecting, which I did not write down unfortunately, where the behavior changed and it froze with a black screen, but this may be irrelevant here.

It freezes and exits without a segfault when I send a SIGTERM to it when frozen.

**Comment by tellowkrinkle** [comment]
Just to be clear

- aaed932a6781264025baa32f4bb56f2cf5146540 goes from
  - [ ] Nothing to freezing and crashing
  - [ ] Freezing to freezing and crashing
  - [ ] Something else
- #5544 goes from
  - [ ] Freezing and crashing to freezing
  - [ ] Freezing to freezing (no change)
  - [ ] Something else

**Comment by NetSysFire** [comment]
aaed932a6781264025baa32f4bb56f2cf5146540 freezes and crashes.
https://github.com/PCSX2/pcsx2/pull/5544 just freezes.

**Comment by tellowkrinkle** [comment]
I'd like to receive enough information to check exactly one of those boxes in each category (sorry, GitHub doesn't have radio buttons)

With the information you've given, I could potentially check all three of the boxes under aaed932a6781264025baa32f4bb56f2cf5146540

**Comment by NetSysFire** [comment]
As I bisected, aaed932a6781264025baa32f4bb56f2cf5146540 introduced the freezing and crashing. It goes from nothing to freezing and crashing. It worked just fine before that commit.
When I close the game window graphically, the game briefly freezes and the emulator then crashes with the trace trap signal (SIGABRT?).
With your PR applied, the emulator just freezes.

Skip this section if it is not relevant. While bisecting, I observed that when compiling 8db0b52821fafabcc556b3669e217bc9c91e5805, the behavior changed. There was no crashing but instead the emulator froze with a black screen. See the bisect log I pasted above. The behavior has since changed since this is an older commit, so this may not be relevant.

**Comment by tellowkrinkle** [comment]
Does the entire program freeze or just some part of it?  Your logs have the main thread in what looks like a gtk event loop, which would make me think it's responding fine
```
Thread 1 (Thread 0x7fb0ba479c00 (LWP 25816) "pcsx2"):
#0  0x00007fb0bde5e2af in poll () at /usr/lib/libc.so.6
#1  0x00007fb0be510066 in  () at /usr/lib/libglib-2.0.so.0
#2  0x00007fb0be4b9463 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#3  0x00007fb0be913bef in gtk_main () at /usr/lib/libgtk-3.so.0
#4  0x00007fb0bf516866 in wxGUIEventLoop::DoRun() () at /usr/lib/libwx_gtk3u_core-3.0.so.0
#5  0x00007fb0bf93a8e2 in wxEventLoopBase::Run() () at /usr/lib/libwx_baseu-3.0.so.0
#6  0x00007fb0bf91445b in wxAppConsoleBase::MainLoop() () at /usr/lib/libwx_baseu-3.0.so.0
#7  0x00007fb0bf978e11 in wxEntry(int&, wchar_t**) () at /usr/lib/libwx_baseu-3.0.so.0
#8  0x000055cbcde2767d in main(int, char**) (argc=<optimized out>, argv=<optimized out>) at /usr/src/debug/pcsx2/pcsx2/gui/AppMain.cpp:72
```

**Comment by NetSysFire** [comment]
Can I test that by switching to the main window where the PCSX2 logo is displayed and I can access settings and boot ISOs etc? If so, this is the case. It does not react to anything, not even the mouse cursor changes shape when trying to enter menus. Is there anything I could provide to debug that part?

Also regarding your fix again, I did some more testing and the crash after the freeze is gone consistently. It appears without the fix it is random (and made me go insane when testing and providing infos) when it will just freeze or it will crash shortly after, but it makes sense it does not always trigger if this is a race condition.

**Comment by NetSysFire** [comment]
Here is a video of me reproducing the freeze of the game window. Apparently that PR also fixed the freeze affecting the main window, I did not notice that before. To be very clear here, the freeze of the main window has been fixed with that PR.

https://user-images.githubusercontent.com/59517351/156079118-43da1cf3-8794-477d-84a2-a5d22cb278a1.mp4

The moment when the animation and sound stops is when I alt+f4'ed the window. The window persists in a broken state but it should be closed instead. The UI continues to work. Closing the main window works as expected.
